In [2]:
import random
sample_space = [1,2,3,4,5,6]
pmf = {1:0.1, 2:0.2, 3:0.2, 4:0.2, 5:0.2, 6:0.1}


# Random Variable Function
def sample():
    return random.choices(sample_space, weights=[pmf[i] for i in sample_space])[0]

samples = [sample() for i in range(1000)]

# Estimate the probability distribution based on samples
estimated_pmf = {i: samples.count(i)/len(samples) for i in sample_space}

print("sample space: ", sample_space)
print("pmf: ", pmf)
print("estimated pmf: ", estimated_pmf)

sample space:  [1, 2, 3, 4, 5, 6]
pmf:  {1: 0.1, 2: 0.2, 3: 0.2, 4: 0.2, 5: 0.2, 6: 0.1}
estimated pmf:  {1: 0.13, 2: 0.189, 3: 0.185, 4: 0.192, 5: 0.197, 6: 0.107}


What is the probability distribution of this function?
Based on the samples, we'd estimate the pmf to be {1: 0.13, 2: 0.189, 3: 0.185, 4: 0.192, 5: 0.197, 6: 0.107}
Based on the actual pmf, it's {1:0.1, 2:0.2, 3:0.2, 4:0.2, 5:0.2, 6:0.1}